src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -37,29 +42,38 @@ const useFetchChannelHook = () => {
       anon_user_info_color_code: channelListInfo?.anonUserInfoColorCode
     };
 
-    channel.targetName = channelListInfo?.channelName;
-    channel.targetImage = channelListInfo?.channelImage;
+    newChannel.targetName = channelListInfo?.channelName;
+    newChannel.targetImage = channelListInfo?.channelImage;
 
-    channel.firstMessage = getFirstMessage(channel?.messages);
+    newChannel.firstMessage = getFirstMessage(channel?.messages);
 
     const isDeletedMessage = channel.firstMessage?.message_type === MESSAGE_TYPE_DELETED;
     if (isDeletedMessage) channel.firstMessage.text = DELETED_MESSAGE_TEXT;
 
-    channel.channel = {...channel};
+    newChannel.channel = {...channel};
     const channelType = channel?.type;
 
     try {
-      const channelList = ChannelList.fromChannelAPI(
-        channel,
-        type[channelType],
-        undefined,
-        anonUserInfo
-      );
-
-      await channelList.saveIfLatest(localDb);
-      refresh('channelList');
+      queue.addJob({
+        label: `saveChannelData-${channel?.id}`,
+        task: () => {
+          return new Promise((resolve) => {
+            const channelList = ChannelList.fromChannelAPI(
+              newChannel,
+              type[channelType],
+              undefined,
+              anonUserInfo
+            );
+
+            channelList.saveIfLatest(localDb).then(() => {
+              refresh('channelList');
+              resolve(true);
+            });
+          });
+        }
+      });
     } catch (e) {
-      console.log('error on helperChannelPromiseBuilder');
+      console.log('error on helperChannelPromiseBuilder', e);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here seem to improve the code by introducing a queue system for managing database operations. This can potentially enhance performance by preventing blocking operations and allowing tasks to be processed asynchronously. However, it's important to handle errors properly within each job task. In the current implementation, if an error occurs during the execution of a job task, it will only be logged and not propagated. Consider adding error handling logic to reject the promise with the error so that it can be handled upstream.
```diff
      queue.addJob({
        label: `saveChannelData-${channel?.id}`,
        task: () => {
          return new Promise((resolve, reject) => {
            const channelList = ChannelList.fromChannelAPI(
              newChannel,
              type[channelType],
              undefined,
              anonUserInfo
            );

            channelList.saveIfLatest(localDb).then(() => {
              refresh('channelList');
              resolve(true);
            }).catch((error) => {
              console.log('Error saving channel list:', error);
              reject(error);
            });
          });
        }
      });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->